### PR TITLE
gh-122361: Use proper `PyUnicodeWriter_*` API in `constevaluator_call`

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-122361

### CODE REVIEW

File: Objects/typevarobject.c [hunk 1]
```diff
@@ -169,38 +169,40 @@ constevaluator_call(PyObject *self, PyObject *args, PyObject *kwargs)
...
-        _PyUnicodeWriter writer;
-        _PyUnicodeWriter_Init(&writer);
+        PyUnicodeWriter *writer = PyUnicodeWriter_Create(5);  // cannot be <5
+        if (writer == NULL) {
+            return NULL;
+        }
         if (PyTuple_Check(value)) {
-            if (_PyUnicodeWriter_WriteASCIIString(&writer, "(", 1) < 0) {
-                _PyUnicodeWriter_Dealloc(&writer);
+            if (PyUnicodeWriter_WriteUTF8(writer, "(", 1) < 0) {
```
Comment (picnixz) [inline Objects/typevarobject.c]: I think we can just use `PyUnicodeWriter_WriteChar` actually

File: Objects/typevarobject.c [hunk 2]
```diff
@@ -169,38 +169,40 @@ constevaluator_call(PyObject *self, PyObject *args, PyObject *kwargs)
...
-                if (_Py_typing_type_repr(&writer, item) < 0) {
-                    _PyUnicodeWriter_Dealloc(&writer);
+                if (_Py_typing_type_repr(writer, item) < 0) {
+                    PyUnicodeWriter_Discard(writer);
                     return NULL;
                 }
             }
-            if (_PyUnicodeWriter_WriteASCIIString(&writer, ")", 1) < 0) {
-                _PyUnicodeWriter_Dealloc(&writer);
+            if (PyUnicodeWriter_WriteUTF8(writer, ")", 1) < 0) {
```
Comment (picnixz) [inline Objects/typevarobject.c]: Ditto.

**Review by picnixz — COMMENTED** [review]
How much is the overhead of using the public API instead of the (slightly, or perhaps more) faster private API?

**Review by JelleZijlstra — APPROVED** [review]
Thanks, thought I got all the warnings.

**Comment by sobolevn** [comment]
> How much is the overhead of using the public API instead of the (slightly, or perhaps more) faster private API?

Oups, sorry, @picnixz, I missed this question! 
I don't really think that it really matters in this use-case :)

**Comment by picnixz** [comment]
No worry (I also don't think it matters (I didn't know in which context it was being used actually))

**Pull request merged**